core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+syntax = "proto3";
+package org.apache.spark.status.protobuf;
+
+message JavaDate {
+  // The number of milliseconds since January 1, 1970, 00:00:00 GMT
+  optional int64 date = 1;
+}
+
+enum JobExecutionStatus {
+  RUNNING = 0;
+  SUCCEEDED = 1;
+  FAILED = 2;
+  UNKNOWN = 3;
+}
+
+message JobData {
+  int32 job_id = 1;
+  string name = 2;
+  optional string description = 3;
+  optional JavaDate submission_time = 4;
+  optional JavaDate completion_time = 5;
```

Comment:
nit: for proto3, we dont need to explicitly specify `optional`.
Any specific reason to call it out ?

Reply:
So that I can use the ".hasXYZ" method to see if a field is set. For example:
https://github.com/apache/spark/pull/38779/files#diff-55f371cbaae6d774066df13f294a74e289a35dd77e30a37ec2663938ecc52b4eR86

Reply:
You dont need `optional` for that IIRC in protobuf 3 iirc.

Reply:
@mridulm I did a quick test and the optional field is needed for `hasFoo`. 
The link you provided seems wrong as per my test.

From https://developers.google.com/protocol-buffers/docs/proto3:
> optional: the same as singular, except that you can check to see if the value was explicitly set.

Reply:
Thanks for checking @gengliangwang ! There might be some specific corner case where that condition applies I guess.
For now, let us go with this itself.